src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -3574,6 +3597,396 @@ void assertReleaseProperties(
+		responseEntityConceptView = this.testRestTemplate.getForEntity(requestUrl, ConceptView.class);
+		concept = responseEntityConceptView.getBody();
+		concept.getDescriptions().add(new Description("Not a real disease"));
+		conceptJson = toJson(concept);
+		requestUrl = "http://localhost:" + port + "/browser/MAIN/projectA/taskC/concepts/" + conceptId;
+		httpEntity = new HttpEntity<>(conceptJson, httpHeaders);
+		this.testRestTemplate.put(requestUrl, httpEntity, void.class);
+
+		// 10. Promote task B
+		mergeReview = buildMergeRequestJson("MAIN/projectA/taskB", "MAIN/projectA");
+		requestUrl = "http://localhost:" + port + "/merges";
+		httpEntity = new HttpEntity<>(mergeReview, httpHeaders);
+		responseEntity = this.testRestTemplate.postForEntity(requestUrl, httpEntity, ResponseEntity.class);
+		jobLocation = responseEntity.getHeaders().get("location").get(0);
+
+		inProgress = true;
+		while (inProgress) {
+			BranchMergeJob branchMergeJob = this.testRestTemplate.getForObject(jobLocation, BranchMergeJob.class);
+			if (branchMergeJob.getStatus() != JobStatus.IN_PROGRESS) {
+				inProgress = false;
+			}
+		}
+
+		// 11. Prepare task C for promotion by rebasing
+		requestUrl = "http://localhost:" + port + "/merge-reviews";
+		mergeReview = buildMergeRequestJson("MAIN/projectA", "MAIN/projectA/taskC");
+		httpEntity = new HttpEntity<>(mergeReview, httpHeaders);
+		responseEntity = this.testRestTemplate.postForEntity(requestUrl, httpEntity, ResponseEntity.class);
+		jobLocation = responseEntity.getHeaders().get("location").get(0);
+
+		inProgress = true;
+		while (inProgress) {
+			MergeReview forObject = this.testRestTemplate.getForObject(jobLocation, MergeReview.class);
+			if (forObject.getStatus() == CURRENT) {
+				inProgress = false;
+			}
+		}
+
+		// 12. Assert there are no conflicts
+		List<?> conflicts = this.testRestTemplate.getForObject(jobLocation + "/details", List.class);
+		assertTrue(conflicts.isEmpty());
+
+		// 13. Assert Concept's Descriptions before rebase
+		requestUrl = "http://localhost:" + port + "/browser/MAIN/projectA/taskC/concepts/" + conceptId;
+		responseEntityConceptView = this.testRestTemplate.getForEntity(requestUrl, ConceptView.class);
+		concept = responseEntityConceptView.getBody();
+		assertEquals(3, concept.getDescriptions().size());
+
+		// 14. Finalise rebase
+		requestUrl = jobLocation + "/apply";
+		httpEntity = new HttpEntity<>(null, httpHeaders);
+		this.testRestTemplate.postForEntity(requestUrl, httpEntity, ResponseEntity.class);
+
+		// 15. Assert Concept's Descriptions after rebase
+		requestUrl = "http://localhost:" + port + "/browser/MAIN/projectA/taskC/concepts/" + conceptId;
+		responseEntityConceptView = this.testRestTemplate.getForEntity(requestUrl, ConceptView.class);
+		concept = responseEntityConceptView.getBody();
+		assertEquals(4, concept.getDescriptions().size()); // Silently merged synonym
+	}
+
+	@Test
+	void testRebaseDescriptionOntoDeletedConcept() {
```

Comment:
This tests the requirements set out in MAINT-1038, i.e. adding synonym to deleted concept.